src/github.com/getlantern/lantern-ui/app/js/app.js

```diff
@@ -122,15 +122,20 @@ var app = angular.module('app', [
           return;
         }
 
-        // try to reconnect indefinitely
-        // when the websocket closes
-        $interval(function() {
-          console.log("Trying to reconnect to disconnected websocket");
-          ds = $websocket('ws://' + document.location.host + '/data');
-          ds.onOpen(function(msg) {
-            $window.location.reload();
-          });
-        }, WS_RECONNECT_INTERVAL);
+        // Temporary workaround for https://bugzilla.mozilla.org/show_bug.cgi?id=1192773
+        if (navigator.userAgent.toLowerCase().indexOf('firefox') > -1) {
+          $rootScope.backendIsGone = true;
+          $rootScope.$digest()
+        } else {
+          // Try to reconnect indefinitely when the websocket closes
+          $interval(function() {
```

Comment:
Yes. My observation was oriented towards the fact that this interval will be created each time this event is triggered and this interval is not saved anywhere so, as far as I understand, it cannot be stopped, if Lantern gets disconnected five times, then we will be running five different intervals checking for the same thing.

It seems like you didn't introduce this so maybe we can follow this problem on a different issue, I think it won't cause I'll effects unless we're running hundreds of timeouts but it may cause some hard to catch bugs.